File: Array_Duplicates/array_duplicates.js

function findDuplicates(arr) {
    // Create an empty array to store duplicate numbers
    let duplicates = [];
    // Loop through each element in the array
    for (let i = 0; i < arr.length; i++) {
      // Check if the current element is present in the remaining elements of the array
      if (arr.slice(i+1).includes(arr[i])) {
        // Check if the current element is not already in the duplicates array
        if (!duplicates.includes(arr[i])) {
          // Add the current element to the duplicates array
          duplicates.push(arr[i]);
        }
      }
    }
    // If there are no duplicates found, return null
    if (duplicates.length === 0) {
      return null;
    }
    // Otherwise, return the duplicates array
    else {
      return duplicates;
    }
  }
  
  // Example usage:
  const arr = [1, 2, 3, 4, 5, 2, 4, 6];
  const duplicates = findDuplicates(arr);
  console.log(duplicates); // Output: [2, 4]
  

File: Array_Duplicates/array_duplicates.py

def find_duplicates(arr):
    # Create an empty list to store duplicate numbers
    duplicates = []
    # Loop through each element in the array
    for i in range(len(arr)):
        # Check if the current element is present in the remaining elements of the array
        if arr[i] in arr[i+1:]:
            # Check if the current element is not already in the duplicates list
            if arr[i] not in duplicates:
                # Add the current element to the duplicates list
                duplicates.append(arr[i])
    # If there are no duplicates found, return None
    if len(duplicates) == 0:
        return None
    # Otherwise, return the duplicates list
    else:
        return duplicates
